[Arch Linux] tthreadmessage.h:87:28: error: ‘Recursive’ is not a member of ‘QMutex’

### Bug Description

The AUR version of the package is very outdated. Since I come from a 1.6 Windows installation I'm not sure I can migrate my projects to the latest AUR working version (1.5.1)
I have tried the AppImage but I'm getting very weird render outputs (i'll open an issue for that as well)
So i tried just building from source myself.
All of the dependencies are installed, I built the bundled tiff and created the systemvar.ini
Whenever it's time to do the make command in tahoma2d/toonz/build/ this happens, I've been hitting my head against it for a week now, I tried Gemini or GH Copilot to help but it has been no use.

```
~/Escritorio/tahoma2d/toonz/build tags/v1.6
❯ make -j$(nproc)
[  0%] Generating __/include/moc_tvectorimage.cpp
[  0%] Generating __/include/moc_tipcsrv.cpp
[  0%] Generating __/include/moc_tmsgcore.cpp
[  0%] Generating __/common/tcore/moc_tthreadp.cpp
[  0%] Generating __/include/moc_tfunctorinvoker.cpp
[  1%] Generating __/include/moc_tipcsrvP.cpp
[  1%] Building C object lzodriver/CMakeFiles/lzocompress.dir/lzocompress.c.o
[  1%] Building C object lzodriver/CMakeFiles/lzodecompress.dir/lzodecompress.c.o
[  1%] Generating __/include/moc_tthread.cpp
[  1%] Generating __/include/moc_tundo.cpp
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/tdebugmessage.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/texception.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/threadmessage.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/tdata.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/tmathutil.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tproperty.cpp.o
[  1%] Linking C executable ../bin/lzocompress
[  1%] Linking C executable ../bin/lzodecompress
[  1%] Built target lzocompress
[  1%] Built target lzodecompress
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/trandom.cpp.o
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/tsmartpointer.cpp.o
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/common/tcore/threadmessage.cpp:3:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:28: error: ‘Recursive’ no es un miembro de ‘QMutex’
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                            ^~~~~~~~~
[  1%] Building CXX object tnzcore/CMakeFiles/tnzcore.dir/__/common/tcore/tstopwatch.cpp.o
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/common/tcore/tsmartpointer.cpp:4:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:28: error: ‘Recursive’ no es un miembro de ‘QMutex’
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                            ^~~~~~~~~
```

Gemini and Copilot both suggested patching the tthreadmessage.h file to change the declarations of "Recursive" there, they worked but eventually showed themselves again in other files. Do I have to build another bundled library? 

### Steps to reproduce

_No response_

### Relevant logs/screenshots

_No response_

### Software Version

1.6

### Software Version Information

_No response_

### Operating System

CachyOS LTS 6.18.22-1 (Arch-based) 

### Graphics Card

_No response_

### Drawing Tablet

_No response_

Ok I checked and i'm using qt 5.15.18 as my main. after doing the cmakelists.txt thing it started building
However now i'm getting this:

```
[ 59%] Building CXX object stdfx/CMakeFiles/tnzstdfx.dir/iwa_bloomfx.cpp.o
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tbigmemorymanager.h:19,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/traster.h:11,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/ttile.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tfx.h:12,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/trasterfx.h:10,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/stdfx.h:7,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowblurfx.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowblurfx.cpp:38:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:37: aviso: ‘QMutex::QMutex(RecursionMode)’ es obsoleto: Use QRecursiveMutex instead of a recursive QMutex [-Wdeprecated-declarations]
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                                     ^
En el fichero incluido desde /usr/include/qt/QtCore/QMutex:1,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:12:
/usr/include/qt/QtCore/qmutex.h:159:14: nota: se declara aquí
  159 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tbigmemorymanager.h:19,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/traster.h:11,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/ttile.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tfx.h:12,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tfxparam.h:8,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_motionflowfx.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_motionflowfx.cpp:1:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:37: aviso: ‘QMutex::QMutex(RecursionMode)’ es obsoleto: Use QRecursiveMutex instead of a recursive QMutex [-Wdeprecated-declarations]
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                                     ^
En el fichero incluido desde /usr/include/qt/QtCore/QMutex:1,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:12:
/usr/include/qt/QtCore/qmutex.h:159:14: nota: se declara aquí
  159 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tbigmemorymanager.h:19,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/traster.h:11,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/ttile.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tfx.h:12,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/trasterfx.h:10,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/stdfx.h:7,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowpaintbrushfx.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowpaintbrushfx.cpp:1:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:37: aviso: ‘QMutex::QMutex(RecursionMode)’ es obsoleto: Use QRecursiveMutex instead of a recursive QMutex [-Wdeprecated-declarations]
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                                     ^
En el fichero incluido desde /usr/include/qt/QtCore/QMutex:1,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:12:
/usr/include/qt/QtCore/qmutex.h:159:14: nota: se declara aquí
  159 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
/home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowblurfx.cpp: In member function ‘virtual void FlowBlurWorker::run()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowblurfx.cpp:183:18: aviso: la iteración 101 invoca comportamiento indefinido [-Waggressive-loop-optimizations]
  183 |       logDist[i] = std::tanh(x / (2.0 * scale));
      |       ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_flowblurfx.cpp:181:23: nota: dentro de este bucle
  181 |     for (int i = 0; i <= 101; i++) {
      |                     ~~^~~~~~
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tbigmemorymanager.h:19,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/traster.h:11,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/ttile.h:6,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tfx.h:12,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/trasterfx.h:10,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/stdfx.h:7,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_bloomfx.h:16,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/stdfx/iwa_bloomfx.cpp:1:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:37: aviso: ‘QMutex::QMutex(RecursionMode)’ es obsoleto: Use QRecursiveMutex instead of a recursive QMutex [-Wdeprecated-declarations]
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                                     ^
En el fichero incluido desde /usr/include/qt/QtCore/QMutex:1,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:12:
/usr/include/qt/QtCore/qmutex.h:159:14: nota: se declara aquí
  159 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
[ 60%] Linking CXX shared library ../lib/tahoma2d/libtnzstdfx.so
[ 60%] Built target tnzstdfx
make: *** [Makefile:136: all] Error 2

```

Nope, that output ends right there without any apparent warning. I tried this:

```
 /Escritorio/tahoma2d/toonz/sources tags/v1.6*

❯ cd ~/Escritorio/tahoma2d/toonz/build

 make -j1

[  8%] Built target tnzcore

[ 12%] Built target tnzbase

[ 14%] Built target tnzext

[ 28%] Built target toonzlib

[ 28%] Built target tfarm

[ 41%] Built target tnzstdfx

[ 41%] Built target sound

[ 42%] Built target colorfx

[ 42%] Building C object image/CMakeFiles/image.dir/tzp/avl.c.o

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:193:7: error: conflicting types for ‘avl__tree’; have ‘TREE *(int,  uint,  int (*)(void *, void *))’ {también conocido como ‘struct avl_tree *(int,  unsigned int,  int (*)(void

*, void *))’}

 193 | TREE *avl__tree(int treetype, uint keyoffs, int (*usrcmp)(void *, void *)) {

     |       ^~~~~~~~~

En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:21:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:226:14: nota: previous declaration of ‘avl__tree’ with type ‘TREE *(int,  unsigned int,  int (*)(void))’ {también conocido como ‘struct avl_tree *(int,  unsigned int,  int (*)(

void))’}

 226 | TNZAPI TREE *avl__tree(int treetype, unsigned keyoffs, int (*usrcmp)());

     |              ^~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__tree’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:247:17: error: assignment to ‘int (*)(void)’ from incompatible pointer type ‘int (*)(void *, void *)’ [-Wincompatible-pointer-types]

 247 |   tree->usrcmp  = usrcmp;

     |                 ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl_insert’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:539:47: error: se pasa el argumento 3 de ‘insert_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

 539 |       ins = insert_ptr(&tree->root, node, tree->usrcmp, NODUP);

     |                                           ~~~~^~~~~~~~

     |                                               |

     |                                               int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:363:29: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 363 |                       int (*usrcmp)(void *, void *), int dup) {

     |                       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘insert_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:542:43: nota: en expansión de macro ‘AVL_AVLCMP’

 542 |       ins = insert_ptr(&tree->root, node, AVL_AVLCMP, NODUP);

     |                                           ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:363:29: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 363 |                       int (*usrcmp)(void *, void *), int dup) {

     |                       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:549:47: error: se pasa el argumento 3 de ‘insert_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

 549 |       ins = insert_ptr(&tree->root, node, tree->usrcmp, DUP);

     |                                           ~~~~^~~~~~~~

     |                                               |

     |                                               int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:363:29: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 363 |                       int (*usrcmp)(void *, void *), int dup) {

     |                       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘insert_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:552:43: nota: en expansión de macro ‘AVL_AVLCMP’

 552 |       ins = insert_ptr(&tree->root, node, AVL_AVLCMP, DUP);

     |                                           ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:363:29: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 363 |                       int (*usrcmp)(void *, void *), int dup) {

     |                       ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__locate’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:669:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 669 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:691:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 691 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__locate_ge’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:738:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 738 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:759:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 759 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__locate_gt’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:795:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 795 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__locate_le’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:831:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 831 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:852:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 852 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__locate_lt’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:888:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

 888 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__remove’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1186:58: error: se pasa el argumento 3 de ‘remove_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

1186 |       node = remove_ptr(&tree->root, (void *)keyval, tree->usrcmp, NODUP);

     |                                                      ~~~~^~~~~~~~

     |                                                          |

     |                                                          int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:972:31: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 972 |                         int (*usrcmp)(void *, void *), int dup) {

     |                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘remove_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1189:54: nota: en expansión de macro ‘AVL_AVLCMP’

1189 |       node = remove_ptr(&tree->root, (void *)keyval, AVL_AVLCMP, NODUP);

     |                                                      ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:972:31: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 972 |                         int (*usrcmp)(void *, void *), int dup) {

     |                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1198:58: error: se pasa el argumento 3 de ‘remove_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

1198 |       node = remove_ptr(&tree->root, (void *)keyval, tree->usrcmp, DUP);

     |                                                      ~~~~^~~~~~~~

     |                                                          |

     |                                                          int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:972:31: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 972 |                         int (*usrcmp)(void *, void *), int dup) {

     |                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘remove_ptr’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1201:54: nota: en expansión de macro ‘AVL_AVLCMP’

1201 |       node = remove_ptr(&tree->root, (void *)keyval, AVL_AVLCMP, DUP);

     |                                                      ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:972:31: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 972 |                         int (*usrcmp)(void *, void *), int dup) {

     |                         ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En el nivel principal:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1242:6: error: conflicting types for ‘avl__scan’; have ‘void(TREE *, void (*)(void *), int)’ {también conocido como ‘void(struct avl_tree *, void (*)(void *), int)’}

1242 | void avl__scan(TREE *tree, void (*usrfun)(void *), int back) {

     |      ^~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:239:13: nota: previous declaration of ‘avl__scan’ with type ‘void(TREE *, void (*)(void), int)’ {también conocido como ‘void(struct avl_tree *, void (*)(void), int)’}

 239 | TNZAPI void avl__scan(TREE *tree, void (*usrfun)(), int back);

     |             ^~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘avl__start’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1374:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

1374 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1436:12: error: assignment to ‘int (*)(void *, void *)’ from incompatible pointer type ‘int (*)(void)’ [-Wincompatible-pointer-types]

1436 |     usrcmp = tree->usrcmp;

     |            ^

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En el nivel principal:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1638:6: error: conflicting types for ‘avl_release’; have ‘void(TREE *, void (*)(void *))’ {también conocido como ‘void(struct avl_tree *, void (*)(void *))’}

1638 | void avl_release(TREE *tree, void (*usrfun)(void *)) {

     |      ^~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:250:13: nota: previous declaration of ‘avl_release’ with type ‘void(TREE *, void (*)(void))’ {también conocido como ‘void(struct avl_tree *, void (*)(void))’}

 250 | TNZAPI void avl_release(TREE *tree, void (*usrfun)());

     |             ^~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c: En la función ‘cdb_decode_all’:

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘avl__tree’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:87:59: nota: en expansión de macro ‘AVL_AVLCMP’

  87 |   avl__tree(AVL_NODUP_INT, avl_offset(structure, member), AVL_AVLCMP)

     |                                                           ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1873:10: nota: en expansión de macro ‘avl_tree_nodup_int’

1873 |   tree = avl_tree_nodup_int(CDB_TREE_ITEM, index);

     |          ^~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:193:51: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 193 | TREE *avl__tree(int treetype, uint keyoffs, int (*usrcmp)(void *, void *)) {

     |                                             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:75:20: error: se pasa el argumento 3 de ‘avl__tree’ desde un tipo de puntero incompatible [-Wincompatible-pointer-types]

  75 | #define AVL_AVLCMP (int (*)())0

     |                    ^~~~~~~~~~~~

     |                    |

     |                    int (*)(void)

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.h:83:59: nota: en expansión de macro ‘AVL_AVLCMP’

  83 |   avl__tree(AVL_NODUP_STR, avl_offset(structure, member), AVL_AVLCMP)

     |                                                           ^~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:1926:23: nota: en expansión de macro ‘avl_tree_nodup_str’

1926 |       item->effects = avl_tree_nodup_str(CDB_EFFECT, name);

     |                       ^~~~~~~~~~~~~~~~~~

/home/ale/Escritorio/tahoma2d/toonz/sources/image/tzp/avl.c:193:51: nota: expected ‘int (*)(void *, void *)’ but argument is of type ‘int (*)(void)’

 193 | TREE *avl__tree(int treetype, uint keyoffs, int (*usrcmp)(void *, void *)) {

     |                                             ~~~~~~^~~~~~~~~~~~~~~~~~~~~~~

make[2]: *** [image/CMakeFiles/image.dir/build.make:485: image/CMakeFiles/image.dir/tzp/avl.c.o] Error 1

make[1]: *** [CMakeFiles/Makefile2:949: image/CMakeFiles/image.dir/all] Error 2

make: *** [Makefile:136: all] Error 2 
```

Almost done. 
Stumbled upon this
```
[ 90%] Building CXX object toonz/CMakeFiles/Tahoma2D.dir/gphotoeventlogpopup.cpp.o
En el fichero incluido desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tbigmemorymanager.h:19,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/traster.h:11,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/toonz/../../sources/stopmotion/gphotocam.h:12,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/toonz/gphotoeventlogpopup.h:7,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/toonz/gphotoeventlogpopup.cpp:1:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h: In constructor ‘TThread::Mutex::Mutex()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:87:37: aviso: ‘QMutex::QMutex(RecursionMode)’ es obsoleto: Use QRecursiveMutex instead of a recursive QMutex [-Wdeprecated-declarations]
   87 |   Mutex() : QMutex(QMutex::Recursive) {}
      |                                     ^
En el fichero incluido desde /usr/include/qt/QtCore/QMutex:1,
                 desde /home/ale/Escritorio/tahoma2d/toonz/sources/include/tthreadmessage.h:12:
/usr/include/qt/QtCore/qmutex.h:159:14: nota: se declara aquí
  159 |     explicit QMutex(RecursionMode mode);
      |              ^~~~~~
/home/ale/Escritorio/tahoma2d/toonz/sources/toonz/gphotoeventlogpopup.cpp: In member function ‘void GPhotoEventLogPopup::onListAllConfigButtonPressed()’:
/home/ale/Escritorio/tahoma2d/toonz/sources/toonz/gphotoeventlogpopup.cpp:239:36: error: ‘class GPhotoCam’ has no member named ‘getCameraAllConfigs’
  239 |   QList<GPConfig> gpConfigs = cam->getCameraAllConfigs();
      |                                    ^~~~~~~~~~~~~~~~~~~
make[2]: *** [toonz/CMakeFiles/Tahoma2D.dir/build.make:4051: toonz/CMakeFiles/Tahoma2D.dir/gphotoeventlogpopup.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1224: toonz/CMakeFiles/Tahoma2D.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```

--- a/toonz/sources/CMakeLists.txt
+++ b/toonz/sources/CMakeLists.txt
@@ -240,7 +240,6 @@ add_definitions(
     -DQT_NETWORK_LIB
     -DQT_CORE_LIB
     -DQT_SHARED
-    -DQT_DISABLE_DEPRECATED_BEFORE=0x050F02
 )
 
 # Find includes in corresponding build directories
@@ -300,6 +299,7 @@ if(BUILD_ENV_MSVC)
     )
     add_definitions(
         -DGLUT_NO_LIB_PRAGMA
+        -DQT_DISABLE_DEPRECATED_BEFORE=0x050F02
     )
 
     if(WITH_GPHOTO2)

--- a/toonz/sources/CMakeLists.txt
+++ b/toonz/sources/CMakeLists.txt
@@ -513,10 +513,15 @@ elseif(BUILD_ENV_UNIXLIKE)
     pkg_check_modules(LZMA liblzma)
     set(TIFF_LIB ${TIFF_LIBRARY} ${LZMA_LIBRARIES})
 
-    if(GLEW-NOTFOUND)
+    if(TARGET GLEW::GLEW)
+        set(GLEW_LIB GLEW::GLEW)
+    elseif(GLEW_FOUND)
+        set(GLEW_LIB ${GLEW_LIBRARIES})
+    else()
+        find_package(PkgConfig REQUIRED)
         pkg_check_modules(GLEW REQUIRED glew)
+        set(GLEW_LIB ${GLEW_LIBRARIES})
     endif()
-    set(GLEW_LIB ${GLEW_LIBRARIES})
 
     pkg_check_modules(LZ4_LIB REQUIRED liblz4)
 

--- a/toonz/sources/image/tzp/avl.h
+++ b/toonz/sources/image/tzp/avl.h
@@ -72,7 +72,7 @@ extern double Avl_Dummy[];
 #define AVL_DUP_USH (AVL_NODUP_USH | 1)
 #define AVL_DUP_CHR (AVL_NODUP_CHR | 1)
 
-#define AVL_AVLCMP (int (*)())0
+#define AVL_AVLCMP (int (*)(void *, void *))0
 
 #define avl_tree_nodup(usrcmp) avl__tree(AVL_NODUP_MBR, (unsigned)0, (usrcmp))
 #define avl_tree_nodup_mbr(structure, member, usrcmp)                          \
@@ -163,7 +163,7 @@ struct avl_path {
 typedef struct avl_tree {
   unsigned short keyinfo;
   unsigned short keyoffs;
-  int (*usrcmp)();
+  int (*usrcmp)(void *, void *);
   long nodes;
   struct avl_node *root;
   struct avl_path *path;
@@ -223,7 +223,7 @@ TNZAPI int avl_porting_problems();
 
 #else
 
-TNZAPI TREE *avl__tree(int treetype, unsigned keyoffs, int (*usrcmp)());
+TNZAPI TREE *avl__tree(int treetype, unsigned keyoffs, int (*usrcmp)(void *, void *));
 TNZAPI TREE *avl_copy(TREE *tree);
 
 TNZAPI int avl_insert(TREE *tree, void *data);
@@ -236,7 +236,7 @@ TNZAPI void *avl_locate_first(TREE *tree);
 TNZAPI void *avl_locate_last(TREE *tree);
 TNZAPI void *avl__remove(TREE *tree, long keyval);
 
-TNZAPI void avl__scan(TREE *tree, void (*usrfun)(), int back);
+TNZAPI void avl__scan(TREE *tree, void (*usrfun)(void *), int back);
 
 TNZAPI void *avl_first(TREE *tree);
 TNZAPI void *avl_last(TREE *tree);
@@ -247,7 +247,7 @@ TNZAPI void avl_stop(TREE *tree);
 
 TNZAPI void *avl__link(TREE *tree, unsigned ptroffs, int back);
 
-TNZAPI void avl_release(TREE *tree, void (*usrfun)());
+TNZAPI void avl_release(TREE *tree, void (*usrfun)(void *));
 TNZAPI void avl_reset(TREE *tree);
 TNZAPI void avl_close(TREE *tree);
 

